dominion-pax-exam-driver/src/main/java/org/codice/dominion/pax/exam/options/extensions/Utilities.java

```diff
@@ -0,0 +1,214 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.dominion.pax.exam.options.extensions;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.annotation.Annotation;
+import java.util.Properties;
+import java.util.function.Function;
+import org.apache.commons.io.IOUtils;
+import org.codice.dominion.options.Options;
+import org.codice.dominion.options.Options.MavenUrl;
+import org.codice.dominion.resources.ResourceLoader;
+import org.codice.test.commons.MavenUtils;
+import org.ops4j.pax.exam.CoreOptions;
+import org.ops4j.pax.exam.options.MavenArtifactUrlReference;
+
+/** Utility methods for defining extensions. */
+public class Utilities {
+  public static boolean isDefined(String value) {
+    return !Options.NOT_DEFINED.equals(value);
+  }
+
+  public static String resolve(String value, String dflt) {
+    return Utilities.isDefined(value) ? value : dflt;
+  }
+
+  public static <T> T applyIfDefined(String value, T t, Function<String, ?> function) {
+    return Utilities.isDefined(value) ? (T) function.apply(value) : t;
+  }
+
+  public static <T, S> T mapAndApplyIfDefined(
+      String value, T t, Function<String, S> mapper, Function<S, ?> function) {
+    return Utilities.isDefined(value) ? (T) function.apply(mapper.apply(value)) : t;
+  }
+
+  public static MavenArtifactUrlReference getProjectReference(
+      Annotation annotation, ResourceLoader resourceLoader) {
+    final Properties dependencies = Utilities.getDependencies(annotation, resourceLoader, null);
+    MavenArtifactUrlReference maven = CoreOptions.maven();
+
+    maven =
```

Comment:
hum! :-) When I first looked at the documentation for that class, I didn't see any and although I noticed it was returning itself, since there was no doc, I didn't want to rely on it. I just noticed that for one, they didn't put any docs and also they didn't put the `@Override` keyword which would have made me look at the base class where it is clearly documented that it returns "self for chaining". Thanks, I prefer chaining.